### PR TITLE
Added Managed Identity authentication 

### PR DESCRIPTION
**Pull request by asaharn**
#### Pull Request Description

_[managed identity](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview) authentication support_

---
**Breaking Changes:**
- None

**Features:**
- Managed Identity authentication

**Fixes:**
- None

Examples:
1) If System-assigned Managed Identity( no need to pass KUSTO_MANAGED_CLIENT_ID):

```
df.write
  .format("com.microsoft.kusto.spark.datasource")
      .format("com.microsoft.kusto.spark.datasource")
      .option(KustoSinkOptions.KUSTO_CLUSTER, "MyCluster")
      .option(KustoSinkOptions.KUSTO_DATABASE, "MyDatabase")
      .option(KustoSinkOptions.KUSTO_TABLE, "MyTable")
      .option(KustoSinkOptions.KUSTO_MANAGED_IDENTITY_AUTH, true.toString)
      .mode(SaveMode.Append)
      .save()
```

2) User-assigned Managed identity:
```
df.write
  .format("com.microsoft.kusto.spark.datasource")
      .format("com.microsoft.kusto.spark.datasource")
      .option(KustoSinkOptions.KUSTO_CLUSTER, "MyCluster")
      .option(KustoSinkOptions.KUSTO_DATABASE, "MyDatabase")
      .option(KustoSinkOptions.KUSTO_TABLE, "MyTable")
      .option(KustoSinkOptions.KUSTO_MANAGED_IDENTITY_AUTH, true.toString)
      .option(KustoSinkOptions.KUSTO_MANAGED_CLIENT_ID, "xxxx-xxxx-xxxxx-xxxx")
      .mode(SaveMode.Append)
      .save()
```

### CODE REVIEW

**Comment by ag-ramachandran** [comment]
Hello @asaharn, couple of quick checks
a) Documentation needs to be added
b) Samples of MI tests can be added in the PR

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoDataSourceUtils.scala [hunk 1]
```diff
@@ -206,7 +209,10 @@ object KustoDataSourceUtils {
       } else if (applicationCertPath.nonEmpty) {
         authentication = AadApplicationCertificateAuthentication(applicationId, applicationCertPath, applicationCertPassword, authorityId)
       }
-    } else if (accessToken.nonEmpty) {
+    }else if( managedIdentityAuth){
```
Comment (ohadbitt) [inline connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoDataSourceUtils.scala:212]: Run ctr+alt+L to format before push

File: connector/src/main/scala/com/microsoft/kusto/spark/common/KustoOptions.scala [hunk 2]
```diff
@@ -49,6 +49,9 @@ trait KustoOptions {
   // The provider will be called for every request to the kusto service
   val KUSTO_TOKEN_PROVIDER_CALLBACK_CLASSPATH: String = newOption("tokenProviderCallbackClasspath")
 
+  val KUSTO_MANAGED_IDENTITY_AUTH : String = newOption("managedIdentityAuth")
+  val KUSTO_MANAGED_CLIENT_ID : String = newOption("managedClientId")
```
Comment (ohadbitt) [inline connector/src/main/scala/com/microsoft/kusto/spark/common/KustoOptions.scala:53]: Managed
```suggestion
  val KUSTO_MANAGED_IDENTITY_CLIENT_ID : String = newOption("managedIdentityClientId")
```

File: docs/Authentication.md [hunk 3]
```diff
@@ -178,4 +178,28 @@ to return an AAD token upon invoking the call method.
 The provider will be called for every request to the kusto service
 
 * **KUSTO_TOKEN_PROVIDER_CALLBACK_CLASSPATH**: 
-    'tokenProviderCallbackClasspath' - The classpath to the class
\ No newline at end of file
+    'tokenProviderCallbackClasspath' - The classpath to the class
+## Managed Identity Authentication
+This authentication method can use [managed identities](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview) to obtain Azure AD tokens without having to manage any credentials.
```
Comment (ohadbitt) [inline docs/Authentication.md:183]: Please explain here how to grant the managed identity access to Kusto

Comment (ohadbitt) [reply docs/Authentication.md:183]: Also refer to this
https://learn.microsoft.com/azure/data-explorer/configure-managed-identities-cluster?tabs=portal

**Review by ohadbitt — APPROVED** [review]

**Pull request merged**